### DISCUSSION TITLE
FilamentAsset::registerScriptData ?

### DISCUSSION THREAD

**Discussion by schlaumeier2023** [filamentphp/filament#9538] [Help]
https://filamentphp.com/docs/3.x/support/assets#registering-script-data

In this section it is written, we can register script data in the boot function of AppServiceProvider, and as an example it is relating to auth()->user() 

However, as my experience in this stage of code, user() is always null and not initializied yet.

**Comment by leandrocfe** [comment]
You are right. You could [send a PR](https://github.com/filamentphp/filament/edit/3.x/packages/support/docs/02-assets.md) to suggest another example

**Reply by schlaumeier2023** [reply]
However, is it a way to share user() as global assest in filamentPHP?

**Reply by sinnbeck** [reply]
Put it in a middleware instead

**Comment by darren-glanville** [comment]
You can get this working, by creating a new Middleware:

```php
<?php

namespace App\Http\Middleware;

use Closure;
use Filament\Support\Facades\FilamentAsset;
use Illuminate\Http\Request;
use Symfony\Component\HttpFoundation\Response;

class JSFilamentData
{
    /**
     * Handle an incoming request.
     *
     * @param  \Closure(\Illuminate\Http\Request): (\Symfony\Component\HttpFoundation\Response)  $next
     */
    public function handle(Request $request, Closure $next): Response
    {
        // Add data to javascript from filament
        FilamentAsset::registerScriptData([
            'user' => auth()->user()->only('id', 'name'),
        ]);

        return $next($request);
    }
}

```

You then need to register the middleware in your panel provider such as **AdminPanelProvider** (depending on if you have multiple ones etc). 

```php
->authMiddleware([
   Authenticate::class,
   \App\Http\Middleware\JSFilamentData::class,
])
```

You need to ensure that your new middleware is **after** the Authenticate class to ensure it is loaded after.

**Comment by dharen008** [comment]
use this on AppServiceProvider > boot()

```php
use Illuminate\Auth\Events\Authenticated;
use Illuminate\Support\Facades\Event;

Event::listen( Authenticated::class, function ( $event ) {
	FilamentAsset::registerScriptData( [
		'user' => [
			'id' => $event->user?->id,
			'name' => $event->user?->name,
		],
	] );
} );
```
